### ISSUE TITLE
kernel: Check potentially interesting new kernel features

### ISSUE THREAD

**Issue by foersleo** [bottlerocket-os/bottlerocket#3201]
\
During the discussions around kernel-6.1 enablement (https://github.com/bottlerocket-os/bottlerocket/pull/3121) a few new kernel features have been mentioned to be potentially interesting to us. Have a look into trade-offs between overhead and benefit of disabling/enabling the following features:

- [ ] FIPS selftest
- [ ] `PREEMPT_DYNAMIC` on arm64
- [ ] `EFI_ZBOOT`
- [ ] `MODULE_DECOMPRESS`
- [ ] `I915` driver
- [ ] `XEN_BACKEND`

**Comment by insanpraja** [comment]
Love to have bottlerocket on XCP-NG.

**Comment by foersleo** [comment]
@insanpraja: Could you elaborate on what you are interested in? As far as I can tell Bottlerocket could make sense as a guest on XCP-NG, which I'd assume should work today (I have not tested that, but Bottlerocket works on xen-based EC2 instances, which I'd assume would be a comparable platform). Is there any specific kernel features that you are missing in order to run Bottlerocket on XCP-NG?

**Comment by insanpraja** [comment]
as per documentation, there are no instruction to deploy or install bottlerocket in XCP-NG, but I suppose we could follow/hack our way through VMware instruction guide?

Thanks!

**Comment by foersleo** [comment]
That is right, the officially supported variants are currently targeting AWS EC2, VMware and a variant for bare metal hosts. Usually the metal variant is the most universal, but needs a tailor made deployment mechanism for whatever target platform.

We are also currently working towards isolating a base image and provide an easy mechanism for the community to build custom versions on top of bottle rocket. Those efforts are tracked in #2669.

If you have a specific ask, feel free to open a new issue. The request for xcp-ng support seems to be not really connected to this issue here. If you have changes to contribute to the project we are also happy to learn about those and work with you to get them integrated.

**Comment by bcressey** [comment]
I'm excited about `EFI_ZBOOT` and poked at it a bit. Some issues:
* no kdump / kexec support
* no x86_64 support - would be nice to always use the same mechanism
* complicated Secure Boot support - need to sign both the "outer" and "inner" images, and `pesign` only signs the outer one
